src/popup/router/pages/Retip.vue

```diff
@@ -54,55 +58,78 @@ export default {
     ...mapGetters(['account', 'tippingSupported', 'minTipAmount', 'activeNetwork']),
-    ...mapState(['balance', 'tippingV1', 'sdk']),
+    ...mapState('fungibleTokens', ['selectedToken']),
+    ...mapState(['balance', 'tippingV1', 'tippingV2', 'sdk']),
     urlStatus() {
       return this.$store.getters['tipUrl/status'](this.tip.url);
     },
+    tippingContract() {
+      return this.$route.query.id.includes('_v2') || this.$route.query.id.includes('_v3')
+        ? this.tippingV2
+        : this.tippingV1;
+    },
   },
   watch: {
     amount() {
-      this.amountError = !+this.amount || this.amount < this.minTipAmount;
+      this.amountError = !+this.amount || (!this.selectedToken && this.amount < this.minTipAmount);
     },
   },
   async created() {
     this.loading = true;
-    await this.$watchUntilTruly(() => this.sdk);
+    await this.$watchUntilTruly(() => this.sdk && this.tippingV1 && this.tippingV2);
     this.minCallFee = calculateFee(TX_TYPES.contractCall, {
       ...this.sdk.Ae.defaults,
-      contractId: this.activeNetwork.tipContractV1,
+      contractId: this.tippingContract.deployInfo.address,
       callerId: this.account.publicKey,
     }).min;
-    await this.$watchUntilTruly(() => this.tippingV1);
-    const tipId = +this.$route.query.id;
+
+    const tipId = this.$route.query.id;
     if (!tipId) throw new Error('"id" param is missed');
-    const { decodedResult } = await this.tippingV1.methods.get_state();
-    this.tip = tipping.getTipsRetips(decodedResult).tips.find(({ id }) => id === tipId);
+    const state = await this.tippingContract.methods.get_state();
+    const tipsAndRetips = await tipping.getTipsRetips(state);
+    this.tip = tipsAndRetips.tips.find(({ id }) => id === tipId);
     this.loading = false;
   },
   methods: {
     async sendTip() {
       const calculatedMaxValue =
         this.balance > this.minCallFee ? this.balance - this.minCallFee : 0;
       this.amountError = !this.amount || !this.minCallFee || calculatedMaxValue - this.amount <= 0;
-      this.amountError = this.amountError || !+this.amount || this.amount < this.minTipAmount;
+      this.amountError =
+        this.amountError ||
+        !+this.amount ||
+        (!this.selectedToken && this.amount < this.minTipAmount);
       if (this.amountError) return;
-      const amount = BigNumber(this.amount).shiftedBy(MAGNITUDE);
+      const amount = convertToken(
+        this.amount,
+        this.selectedToken ? this.selectedToken.decimals : MAGNITUDE,
+      ).toFixed();
       this.loading = true;
       try {
-        const { hash } = await this.tippingV1.methods.retip(this.tip.id, {
-          amount,
-          waitMined: false,
-          modal: false,
-        });
-        if (hash) {
-          this.$store.commit('addPendingTransaction', {
-            hash,
-            amount: this.amount,
-            domain: this.tip.url,
-            type: 'tip',
+        let retipResponse = null;
+        if (this.selectedToken) {
```

Comment:
Does retipping with fungible tokens work for the tips from v1 contract? If not then we shouldn't allow it in the ui

Reply:
Good catch. It should only work with v2/v3 contracts. Added a validation with error message for such cases.